pkg/kubelet/util/csr/csr.go

```diff
@@ -50,67 +60,136 @@ func RequestNodeCertificate(client certificatesclient.CertificateSigningRequestI
 	if err != nil {
 		return nil, fmt.Errorf("unable to generate certificate request: %v", err)
 	}
-	return RequestCertificate(client, csrData, []certificates.KeyUsage{
+
+	usages := []certificates.KeyUsage{
 		certificates.UsageDigitalSignature,
 		certificates.UsageKeyEncipherment,
 		certificates.UsageClientAuth,
-	})
-}
+	}
 
-// RequestCertificate will create a certificate signing request using the PEM
-// encoded CSR and send it to API server, then it will watch the object's
-// status, once approved by API server, it will return the API server's issued
-// certificate (pem-encoded). If there is any errors, or the watch timeouts, it
-// will return an error.
-func RequestCertificate(client certificatesclient.CertificateSigningRequestInterface, csrData []byte, usages []certificates.KeyUsage) (certData []byte, err error) {
```

Comment:
How hard is it to leave requestCertificate as a function? There is an exact copy of this in the certificate_manager code that only exists because of conflicting imports.

https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/certificate/certificate_manager.go#L345-L354

Hopefully in 1.8 kubelet will transition to new imports and there would be an opportunity to eliminate this duplication.